### PR TITLE
Add async sink.

### PR DESCRIPTION
**Pull request by PremyslCerny**

### CODE REVIEW

**Comment by PremyslCerny** [comment]
@ateska Please take a look.

File: bspump/aio/sink.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+		self.QueueMaxSize = int(self.Config["output_queue_max_size"])
+		assert (self.QueueMaxSize >= 1)
+
+		self.Task = asyncio.ensure_future(
+			self._task()
+		)
+
+		self.Exiting = False
+
+		app.PubSub.subscribe("Application.exit!", self._on_exit)
```
Comment (ateska) [inline bspump/aio/sink.py:41]: There should be automatic Task restart when exception is raised in `consume()` + good logging of that exception + throttling of the pipeline

**Review by ateska — CHANGES_REQUESTED** [review]
Looks nice.

File: bspump/aio/sink.py (hunk 1 above)
Comment (PremyslCerny) [reply bspump/aio/sink.py:41]: @ateska The exception handling is already there, there is a while loop in the task with try catch inside.

**Review by ateska — APPROVED** [review]

File: bspump/aio/sink.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+				await self.consume(context, event)
+
+			except Exception as e:
+				L.exception("The following exception occurred in async sink: '{}'".format(e))
+
+
+	async def _on_exit(self, event_name):
+		self.Exiting = True
+
+		if self.Task is not None:
```
Comment (ateska) [inline bspump/aio/sink.py:65]: Consider calling `.cancel()` on the task.

**Review by ateska — CHANGES_REQUESTED** [review]

**Pull request merged**